src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/prefermethodreference/InputPreferMethodReferenceObjectCreation.java

```diff
@@ -0,0 +1,39 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.prefermethodreference;
+
+import java.util.ArrayList;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Config = default
+ * Object/Array creation cases
+ */
+public class InputPreferMethodReferenceObjectCreation {
+    int field;
+    // object creation
+    Supplier<Object> s = () -> new Object(); // violation
```

Comment:
please add test cases for (with and without curly braces):
```
() -> { return new HashSet<>(); }
() -> { return new HashSet<String>(); }
```
to be sure that Check behavior is not affected by generics.

Reply:
done